depthai_ros_driver/include/depthai_ros_driver/converter_and_stamper.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+/**
+ * @file converter_and_stamper.hpp
+ * @brief Copy the stamp from one topic and convert an unstamped topic into a stamped topic
+ */
+#include <cstdint>
+
+#include <ros/message_traits.h>
+#include <ros/ros.h>
+#include <ros/time.h>
+#include <topic_tools/shape_shifter.h>
+
+namespace rr {
+/**
+ * @brief Finds the length of the header without deserializing it or performing any checks
+ *
+ * @param data pointer to the serialized data
+ * @return std::size_t length of the header
+ */
+std::size_t find_header_length(const std::uint8_t* const data) {
+    std::size_t count = 0;
+
+    const auto& len = [](auto t) { return ros::serialization::Serializer<decltype(t)>::serializedLength(t); };
```

Comment:
Will having `constexpr` boost up this function? I feel like len(std::uint32_t{}) can definitely use constexpr.

Reply:
I don't think it's huge impact. I can make part of the header size calculation static, to eliminate repeated calculation impact